tests/integration/test_experiments.py

```diff
@@ -4,13 +4,31 @@ def test_list(client, create_experiment):
     for name in expected_experiments:
         create_experiment(name)
 
-    resp = client.get_check(f"/experiments")
-    assert expected_experiments == {exp["name"] for exp in resp.json()}
+    docs = [
+        exp
+        for exp in client.get_check("/experiments").json()
+        if exp["name"] in expected_experiments
+    ]
+
+    assert expected_experiments == {exp["name"] for exp in docs}
+    assert all(exp["ensembles"] == [] for exp in docs)
 
 
 def test_ensembles(client, create_experiment, create_ensemble):
-    nr_of_ensembles = 5
     experiment_id = create_experiment("test_ensembles")
-    ids = {create_ensemble(experiment_id) for i in range(nr_of_ensembles)}
+    ids = {create_ensemble(experiment_id) for _ in range(5)}
+
+    # Ensembles exist when using experiment list endpoint
+    docs = client.get_check("/experiments").json()
+    for doc in docs:  # Python doesn't have a built-in find function :(
+        if doc["id"] == experiment_id:
+            break
+    else:
+        raise KeyError(
+            f"Experiment with id '{experiment_id}' not found in the list of experiments"
+        )
+    assert ids <= set(doc["ensembles"])
+
+    # The list of ensembles belonging to the newly created experiment matches
     resp = client.get_check(f"/experiments/{experiment_id}/ensembles")
-    assert ids == {ens["id"] for ens in resp.json()}
+    assert ids <= {ens["id"] for ens in resp.json()}
```

Comment:
why less or equal?

Reply:
`<=` means `ids` is a subset of the IDs that exist on the database. This is to have the tests pass when there's data already in the database.